### ISSUE TITLE
Check client connection status in Server

### ISSUE THREAD

**Issue by mohammadnejad** [marcoferrer/kroto-plus#91]
Hello, In bidirectional streaming How I can check client connection status? I am developing multiplayer online game with your library and I want to know when one player disconnect and notify to other players?

**Comment by marcoferrer** [comment]
I’ll be able to give you a more explicit answer a little later. But in the mean time this example might be of some use. It’s a demo of a multi-player game for a talk. https://github.com/leveretka/grpc-death-star/pull/3

**Comment by mohammadnejad** [comment]
Yes, when client disconnects I handle cancellation notification and everything is OK, but when internet in client disconnects, client can't send cancellation notification to server and I can't handle it.

**Comment by marcoferrer** [comment]
When the network disconnects the grpc server should invoke the cancellation callback as well. I havent tried it explicitly but it is mentioned in [ServerCall.onCancel()](https://github.com/grpc/grpc-java/blob/80c3c992a66aa21ccf3e12e38000316e45f97e64/api/src/main/java/io/grpc/ServerCall.java#L75)

**Comment by mohammadnejad** [comment]
I handle the problem with config keepalivetime and keepalivetimout. Server ping client every 20 seconds and when the network disconnects the server close stream.
